### ISSUE TITLE
Low Audio Quality

### ISSUE THREAD

**Issue by RGBCube** [z-huang/music#1294]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/z-huang/InnerTune/actions).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce the bug

Listen to any song in InnerTune and then on [ViMusic](https://github.com/vfsfitvnm/ViMusic) and notice the quality difference.

### Expected behavior

Better Quality music.

### Actual behavior

Low and murky quality music.

### InnerTune version

0.5.3

### Android version

12

### Additional information

Even when the Audio Quality setting is set to High, InnerTune still fetches murky audio. I have tried the same song in [ViMusic](https://github.com/vfsfitvnm/ViMusic), and there it is definitely higher quality. Maybe the YouTube Music API is giving InnerTune low quality music because of a user agent or any identifiable information?

This could be fixed by copying the ViMusic downloader implementation over with minimal changes. Hopefully then audio quality will be better for High Quality mode.

**Comment by RGBCube** [comment]
Could the low quality be related with logging into one's YouTube Music account? Because after logging out the quality is no longer low. Weird?

**Comment by RGBCube** [comment]
Could the low quality be related with logging into one's YouTube Music account? Because after logging out the quality is no longer low. Weird?

**Comment by RGBCube** [comment]
Actually nevermind, it's low even without logging in now

**Comment by z-huang** [comment]
InnerTune gets the highest quality possible if you set quality to high. The highest quality is determined by YouTube. Can you provide information about the quality difference between InnerTune and ViMusic?

**Comment by RGBCube** [comment]
> InnerTune gets the highest quality possible if you set quality to high. The highest quality is determined by YouTube. Can you provide information about the quality difference between InnerTune and ViMusic?

You are able to hear it, it sounds murky compared to ViMusic. The YouTube API might be checking for user agents or anything that signals that the client is not authentic and if it isn't, it might be serving lower quality music to them (only sometimes, for some reason).

This doesn't happen with ViMusic, could you check what it does differently (compared to InnerTune) which lets it get the highest quality music all the time?

**Comment by z-huang** [comment]
Statistically, InnerTune and ViMusic can both get 140~150kbps quality. Methodologically, InnerTune and ViMusic use a very similar method. You may need to provide more information, thanks.

**Comment by darkza** [comment]
is there any option like "statistics for interested" so that you can see what quality actually is playing?

**Comment by z-huang** [comment]
In the player screen, click "..." at the bottom, and click "Details".

**Comment by RGBCube** [comment]
> In the player screen, click "..." at the bottom, and click "Details".

![Screenshot_20240808-192701_InnerTune.png](https://github.com/user-attachments/assets/4555076f-a4ac-48fc-a866-3dab741bee3d)

![Screenshot_20240808-192643_InnerTune.png](https://github.com/user-attachments/assets/68d7e54b-06a0-49f7-ae67-719232acc84f)

![Screenshot_20240808-192714_InnerTune.png](https://github.com/user-attachments/assets/1df8c2b3-4bd8-48b9-bf83-5351dbb1ce6b)


Most of the time I get around 50kbps, only rarely 150ish.

However on ViMusic I always get the highest. Are you sure all the info served to the YouTube API is similar between ViMusic and InnerTune?

**Comment by frutejuise** [comment]
On the same song I get 134kbps, 50s is extremely low
![Screenshot_20240814-215934.png](https://github.com/user-attachments/assets/db86a830-8cb1-4ff6-8e2b-df9d736a92cb)

**Comment by frutejuise** [comment]
![Screenshot_20240814-220137.png](https://github.com/user-attachments/assets/fbfa22dd-16de-4ede-84ad-fa717b62e294)

**Comment by frutejuise** [comment]
I don't think I've ever had a problem with the quality, my only thought is yours could be on auto?

**Comment by RGBCube** [comment]
> I don't think I've ever had a problem with the quality, my only thought is yours could be on auto?

nope, just checked and it is on High. I also have Audio Normalization enabled, but that shouldn't affect anything should it?

**Comment by frutejuise** [comment]
Not sure.

**Comment by RGBCube** [comment]
Hmm, I think I've figured it out. Something is persisting that makes InnerTune fetch low quality audio when you listen to a song on auto/low quality, then switch to high. Clearing the audio cache doesn't fix this, which is the problem here. @z-huang could you look into this? Should be an easy fix.

**Comment by ganairm** [comment]
i have the same issue... after trying to log in to youtube music...  it was working perfectly fine, but then as soon as i logged in, some songs are fetched in lower bit-rate

**Comment by gechoto** [comment]
For those stuck with low audio quality I described a workaround here:
https://github.com/z-huang/InnerTune/issues/1631#issuecomment-2433388229
